src/eleveldb_metadata.erl

```diff
@@ -0,0 +1,110 @@
+%% -------------------------------------------------------------------
+%%
+%%  eleveldb_metadata: Erlang interface to allow LevelDB to retrieve riak_core metadata
+%%
+%% Copyright (c) 2017 Basho Technologies, Inc. All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(eleveldb_metadata).
+
+-export([start_link/0]).
+
+-export([callback_router/3]).
+
+%% Application messages are sent to this process directly from
+%% NIF-land instead of other Erlang modules, so this does not
+%% implement an OTP behavior.
+%%
+%% See http://erlang.org/doc/design_principles/spec_proc.html for
+%% information on these system message handlers.
+-export([system_continue/3, system_terminate/4,
```

Comment:
If this were a `gen_whatever` shutdown messages could unregister the service, as I've suggested elsewhere.